src/restconfig/src/main/java/org/geoserver/rest/resources/ResourceResource.java

```diff
@@ -0,0 +1,409 @@
+        public List<ResourceChild> getChildren() {
+            return children;
+        }
+    }
+    
+    private ResourceStore store;
+    private Resource resource;
+    private Operation operation;
+    
+    public ResourceResource(Context context, Request request, Response response, Resource resource, Operation operation,
+            ResourceStore store) {
+        super(context, request, response);
+        this.resource = resource;
+        this.operation = operation;
+        this.store = store;
+    }
+    
+    public Resource getResource() {
+        return resource;
+    }
+
+    public Operation getOperation() {
+        return operation;
+    }
+        
+    private String href(String path, boolean isDir) {
+        String href = getPageInfo().rootURI("resources/" + path);
+        if (isDir) {
+            String format = (String) getRequest().getAttributes().get("format");
+            if (format != null) {
+                href += "?format=" + format;
+            }
+        }
+        return href;
+    }
+    
+    public String formatHtmlLink(String link) {
+        return link.replaceAll("&", "&amp;");
+    }
+
+    private MediaType getMediaType(Resource resource) {
+        if (resource.getType() == Type.DIRECTORY) {
+            return getFormatGet(false).getMediaType();
+        } else if (resource.getType() == Type.RESOURCE) {
+            String mimeType = URLConnection.guessContentTypeFromName(resource.name());
+            if (mimeType == null || MediaType.APPLICATION_OCTET_STREAM.getName().equals(mimeType)) {
+                //try guessing from data
+                try (InputStream is = new BufferedInputStream(resource.in())) {
+                    mimeType = URLConnection.guessContentTypeFromStream(is);
+                } catch (IOException e) {
+                    //do nothing, we'll just use application/octet-stream
+                }
+            }
+            return mimeType == null ? MediaType.APPLICATION_OCTET_STREAM : MediaType.valueOf(mimeType);
+        } else {
+            return null;
+        }
+    }
+    
+    @Override
+    protected List<DataFormat> createSupportedFormats(Request request, Response response) {
+        List<DataFormat> formats = new ArrayList<DataFormat>();
+        
+        //HTML
+        if (operation == Operation.METADATA || resource.getType() == Type.DIRECTORY) {
+            formats.add(new ReflectiveHTMLFormat(request,response,this) {   
+                @Override
+                protected Configuration createConfiguration(Object data, Class clazz) {
+                    Configuration cfg = super.createConfiguration(data, clazz);
+                    cfg.setClassForTemplateLoading(ResourceResource.class, "");
+                    cfg.setObjectWrapper(new ObjectToMapWrapper<ResourceMetadata>(ResourceMetadata.class) {
+                        @Override
+                        protected void wrapInternal(Map properties, SimpleHash model, ResourceMetadata object) {
```

Comment:
I always get rid of all warnings when I submit new code, however I have "ignore" set on unavoidable raw types (because of superclasses using raw types). Add the @SuppressWarnings then introduces a warning (Unnecessary @SuppressWarnings("unchecked") 
Personally I always use generics and try to only use the suppresswarnings("unchecked") when I am forced to do a class cast that I know if safe but eclipse doesn't.